pipeline/src/test_util.rs

```diff
@@ -58,13 +60,36 @@ pub fn verify_pipeline(pipeline: Pipeline<GoldilocksField>) -> Result<(), String
 
 pub fn gen_estark_proof(file_name: &str, inputs: Vec<GoldilocksField>) {
     let tmp_dir = mktemp::Temp::new_dir().unwrap();
-    Pipeline::default()
+    let mut pipeline = Pipeline::default()
         .with_tmp_output(&tmp_dir)
         .from_file(resolve_test_file(file_name))
         .with_prover_inputs(inputs)
-        .with_backend(powdr_backend::BackendType::EStark)
-        .compute_proof()
-        .unwrap();
+        .with_backend(powdr_backend::BackendType::EStark);
+
+    pipeline.clone().compute_proof().unwrap();
+
+    // Repeat the proof generation, but with an externally generated verification key
+    let pil = pipeline.compute_optimized_pil().unwrap();
+
+    // Verification Key
+    let vkey_file_path = tmp_dir.as_path().join("verification_key.bin");
+    let vkey_file = BufWriter::new(File::create(&vkey_file_path).unwrap());
+    write_or_panic(vkey_file, |writer| {
+        pipeline.export_verification_key(writer).unwrap()
+    });
+
+    // Create the proof before adding the vkey to the pipeline,
+    // so that it's generated during the proof
```

Comment:
Because it should re-generate the same one we're adding?

Reply:
yep